pkg/kubectl/cmd/set/set_limit.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+)
+
+// LimitOptions is the start of the data required to perform the operation. As new fields are added, add them here instead of
+// referencing the cmd.Flags
+type LimitOptions struct {
+	Mapper            meta.RESTMapper
+	Typer             runtime.ObjectTyper
+	Infos             []*resource.Info
+	Encoder           runtime.Encoder
+	Out               io.Writer
+	Selector          string
+	Err               io.Writer
+	Filenames         []string
+	ContainerSelector string
+	Recursive         bool
+	ShortOutput       bool
+	All               bool
+	Record            bool
+	ChangeCause       string
+	Local             bool
+	Cmd               *cobra.Command
+
+	Limits               string
+	Requests             string
+	Remove               bool
+	ResourceRequirements api.ResourceRequirements
+
+	PrintObject            func(cmd *cobra.Command, mapper meta.RESTMapper, obj runtime.Object, out io.Writer) error
+	UpdatePodSpecForObject func(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error)
+	Resources              []string
+}
+
+const (
+	limit_resources = `
+  pod (po), replicationcontroller (rc), deployment, daemonset (ds), job, replicaset (rs)`
```

Comment:
Note that that method should *not* be making a dynamic call to retrieve these, it should just be a list of constants.  Otherwise, help would not work if the user can't access the network.